akka-actor-typed-tests/src/test/scala/akka/actor/typed/LocalActorRefProviderLogMessagesSpec.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.testkit.typed.scaladsl.{ ActorTestKit, LoggingTestKit, ScalaTestWithActorTestKit }
+import akka.actor.typed.internal.adapter.ActorSystemAdapter
+import org.scalatest.wordspec.AnyWordSpecLike
+
+object LocalActorRefProviderLogMessagesSpec {
+  val config = """
+    akka {
+      loglevel = DEBUG # test verifies debug
+      log-dead-letters = on
+      actor {
+        debug.unhandled = on
+      }
+    }
+  """
+}
+
+class LocalActorRefProviderLogMessagesSpec
+    extends ScalaTestWithActorTestKit(LocalActorRefProviderLogMessagesSpec.config)
+    with AnyWordSpecLike {
+
+  "An LocalActorRefProvider" must {
+
+    "logs on dedicated 'serialization' logger of unknown path" in {
+      val provider = system.asInstanceOf[ActorSystemAdapter[_]].provider
+
+      LoggingTestKit
+        .debug("of unknown (invalid) path [dummy/path]")
+        .withLoggerName("akka.actor.LocalActorRefProvider.Deserialization")
+        .expect {
+          provider.resolveActorRef("dummy/path")
+        }
+    }
+
+    "logs on dedicated 'serialization' logger when path doesn't match existing actor" in {
+      val provider = system.asInstanceOf[ActorSystemAdapter[_]].provider
+      val invalidPath = provider.rootPath / "user" / "invalid"
+
+      LoggingTestKit
+        .debug("Resolve (deserialization) of path [user/invalid] doesn't match an active actor.")
+        .withLoggerName("akka.actor.LocalActorRefProvider.Deserialization")
+        .expect {
+          provider.resolveActorRef(invalidPath)
+        }
+    }
+
+    "logs on dedicated 'serialization' logger when of foreign path" in {
+
+      val otherSystem = ActorTestKit("otherSystem").system.asInstanceOf[ActorSystemAdapter[_]]
+      val invalidPath = otherSystem.provider.rootPath / "user" / "foo"
+
+      val provider = system.asInstanceOf[ActorSystemAdapter[_]].provider
+      try {
+        LoggingTestKit
+          .debug("Resolve (deserialization) of foreign path [akka://otherSystem/user/foo]")
+          .withLoggerName("akka.actor.LocalActorRefProvider.Deserialization")
+          .expect {
+            provider.resolveActorRef(invalidPath)
+          }
+      } finally {
+        ActorTestKit.shutdown(otherSystem)
+      }
+    }
+
+    // "logs on dedicated 'serialization' logger for empty paths" in {
+
+    //   val provider = system.asInstanceOf[ActorSystemAdapter[_]].provider
+    //   val invalidPath = provider.rootPath
+
+    //   LoggingTestKit
+    //     .debug("Resolve (deserialization) of empty path")
+    //     .withLoggerName("akka.actor.LocalActorRefProvider.Deserialization")
+    //     .expect {
+    //       provider.resolveActorRef(invalidPath.toString)
+    //     }
+    // }
```

Comment:
This test doesn't work because the `PathUtils` (https://github.com/akka/akka/blob/actor-ref-provider-logging/akka-actor/src/main/scala/akka/actor/Address.scala#L134-L136) always produces a non-empty list.

When `sub` is empty, it should not add it to `acc`. As it is now, it returns a `List("")`.

I'm looking further if a fix will break other tests and if that can be an issue if left as such.